workflow/src/test/java/com/google/android/fhir/workflow/activity/ActivityFlowTest.kt

```diff
@@ -0,0 +1,782 @@
+        }
+      }
+      .onFailure { fail("Unexpected", it) }
+
+    val planToResume =
+      repository
+        .read(MedicationRequest::class.java, IdType("MedicationRequest", cachedRequestId))
+        .let { CPGMedicationRequest(it) }
+    val resumedPlanFlow = ActivityFlow.of(repository, planToResume)
+
+    check(resumedPlanFlow.getCurrentPhase() is PlanPhase<*>) {
+      "Flow is in ${resumedPlanFlow.getCurrentPhase().getPhaseName()} "
+    }
+    resumedPlanFlow
+      .prepareOrder()
+      .onSuccess { draftOrder ->
+        draftOrder.update { addNote(Annotation(MarkdownType("Draft order looks OK."))) }
+
+        resumedPlanFlow.initiateOrder(draftOrder).onSuccess { orderPhase ->
+          val updatedOrder =
+            orderPhase.getRequestResource().copy().apply {
+              setStatus(Status.ACTIVE)
+              update { addNote(Annotation(MarkdownType("Order looks OK."))) }
+            }
+
+          orderPhase
+            .update(updatedOrder)
+            .onSuccess { cachedRequestId = updatedOrder.logicalId }
+            .onFailure { fail("Should have succeeded", it) }
+        }
+      }
+      .onFailure { fail("Unexpected", it) }
+
+    val orderToResume =
+      repository
+        .read(MedicationRequest::class.java, IdType("MedicationRequest", cachedRequestId))
+        .let { CPGMedicationRequest(it) }
+
+    val resumedOrderFlow = ActivityFlow.of(repository, orderToResume)
+
+    check(resumedOrderFlow.getCurrentPhase() is OrderPhase<*>) {
+      "Flow is in ${resumedOrderFlow.getCurrentPhase().getPhaseName()} "
+    }
+
+    resumedOrderFlow
+      .preparePerform(CPGMedicationDispenseEvent::class.java)
+      .onSuccess { draftEvent ->
+        draftEvent.update { addNote(Annotation(MarkdownType("Draft event looks OK."))) }
+
+        resumedOrderFlow.initiatePerform(draftEvent).onSuccess { performPhase ->
+          val updatedEvent =
+            performPhase.getEventResource().copy().apply {
+              setStatus(EventStatus.INPROGRESS)
+              update { addNote(Annotation(MarkdownType("Event looks OK."))) }
+            }
+
+          performPhase
+            .update(updatedEvent)
+            .onSuccess { cachedRequestId = updatedEvent.logicalId }
+            .onFailure { fail("Should have succeeded", it) }
+        }
+      }
+      .onFailure { fail("Unexpected", it) }
+
+    val eventToResume =
+      repository
+        .read(MedicationDispense::class.java, IdType("MedicationDispense", cachedRequestId))
+        .let { CPGMedicationDispenseEvent(it) }
+
+    val resumedPerformFlow = ActivityFlow.of(repository, eventToResume)
+
+    check(resumedPerformFlow.getCurrentPhase() is Phase.EventPhase<*>) {
+      "Flow is in ${resumedPerformFlow.getCurrentPhase().getPhaseName()} "
+    }
+
+    (resumedPerformFlow.getCurrentPhase() as Phase.EventPhase<*>).complete()
+  }
+
+  @Test
+  fun `draftPlan is success when flow is in proposal phase`(): Unit = runBlockingOnWorkerThread {
```

Comment:
Maybe this is better ?
```suggestion
  fun `preparePlan should succeed when in proposal phase`(): Unit = runBlockingOnWorkerThread {
```

Reply:
Done